app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/DeclarationJourneyController.scala

```diff
@@ -36,7 +36,7 @@ trait DeclarationJourneyUpdateController extends DeclarationJourneyController {
 trait IndexedDeclarationJourneyController extends FrontendBaseController {
   implicit def messages(implicit request: Request[_]): Messages = controllerComponents.messagesApi.preferred(request)
 
-  def onPageLoad(idx: Int): Action[AnyContent]
+  def onPageLoad(idx: Int, change: Boolean): Action[AnyContent]
```

Comment:
This will work for now but only until we need to make the /check-your-answers page 'change' links work too.

We could pass something other than a Boolean, perhaps a NavigationMode enum with values [Normal, ReviewGoods] for now that can be extended to include 'CheckYourAnswers'?

Another way might be not to use a query parameter but instead to check whether the GoodsEntry is complete (i.e. we can create a Goods from it) and if it is then we redirect to /review-goods. 

I think we will need to do this for the change links on /check-your-answers as the navigation after a change will need to be conditional. For example if you've previously entered 'no' to 'are you a customs agent?' then go back to change this to 'yes' you'll need to be redirected as normal the /agent-details rather than back to /check-your-answers.

There are other ways to achieve this of course...